### DISCUSSION TITLE
How to disable a TabPane?

### DISCUSSION THREAD

**Discussion by grantm** [Textualize/textual#3116] [Q&A] [answered]
I'm very new to `textual` - it seems very cool so far. I've set up an app with this structure:
```
    def compose(self):
        yield Header()
        with TabbedContent():
            with TabPane("Settings", id="settings-pane"):
                yield Static("Settings stuff goes here")
            with TabPane("Search", id="search-pane", disabled=True, classes="needs-token"):
                yield Static("Search stuff goes here")
            with TabPane("Browse", id="browse-pane", disabled=True, classes="needs-token"):
                yield Static("Browse stuff goes here")
        yield Footer()

```
I thought that `disabled=True` would mean that the user could not activate the `TabPane` but that doesn't seem to be the case.
Then I thought I'd try using CSS to hide them instead:
```
TabPane.needs-token {
    display: none;
    background: red;
}
```
The text for both the second and the third `TabPane` is displayed with a red background, so the selector obviously matches, but the `display: none` seems to have no effect.
What am I missing?

**Comment by TomJGooding** [comment, marked as answer]
Disabling tabs is in the works - see #3088.

**Comment by davep** [comment]
This isn't supported right now, but is currently being worked on: https://github.com/Textualize/textual/issues/3088

**Reply by TomJGooding** [reply]
Oops, sorry we must have posted within seconds of each other.

**Reply by davep** [reply]
Heh! Too many right answers > no answers. ;-)

**Comment by grantm** [comment]
Thanks for the prompt response. I had searched the open issues but somehow failed to find that one.
I now have a workaround using CSS selectors that set the children of the `TabPane` to `display: none;` so it's not a show stopper.
Thanks again.
